### ISSUE TITLE
Defer loading of extensions

### ISSUE THREAD

**Issue by brichet** [jupyterlab/jupyterlab#14576]
### Problem

Some extensions can be very long to load, and extends the pageload time of the application.
It could be interesting to defer the loading of some extensions after the application is started.

In addition, there may be a confusion between `autoStart` option and `deferredExtensions` attribute:
> Currently the *page_config.json* allows a `deferredExtensions` attribute, containing a list of extensions or plugins. 
> It looks like these deferred extension are not started at all if they are not required. It acts like the `autoStart: false` plugin parameter.

### Proposed Solution

1. **Lumino**
- add a `deferred` option (boolean) to the plugins, to start a plugin only after the main application is started.

2. **JupyterLab**
- use the `deferredExtensions` attribute from *page_config.json* like the deferred option from plugins.
- wait for the application fully restored to activate the deferred extensions.
- *OPTIONAL*: add an `autoStartExtensions` attribute to the *page_config.json*, which act like the `autoStart` option in plugins.
